lib/tasks/poller.rake

```diff
@@ -0,0 +1,84 @@
+# BigBlueButton open source conferencing system - http://www.bigbluebutton.org/.
+#
+# Copyright (c) 2022 BigBlueButton Inc. and by respective authors (see below).
+#
+# This program is free software; you can redistribute it and/or modify it under the
+# terms of the GNU Lesser General Public License as published by the Free Software
+# Foundation; either version 3.0 of the License, or (at your option) any later
+# version.
+#
+# Greenlight is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with Greenlight; if not, see <http://www.gnu.org/licenses/>.
+
+# frozen_string_literal: true
+
+namespace :poller do
+  # Does a check if a meeting set as online is still online
+  task :meetings_poller, %i[provider] => :environment do |_task, args|
+    args.with_defaults(provider: 'greenlight')
+
+    online_meetings = Room.joins(:user)
+                          .where(users: { provider: args[:provider] }, online: true)
+
+    RunningMeetingChecker.new(rooms: online_meetings, provider: args[:provider]).call
+
+  rescue StandardError => e
+    err "Unable to poll meetings. Error: #{e}"
+  end
+
+  # Does a check if a recording from a recent meeting has not been created in GL
+  task :recordings_poller, %i[provider] => :environment do |_task, args|
+    args.with_defaults(provider: 'greenlight')
+
+    recent_meetings = Room.joins(:user)
+                          .where(users: { provider: args[:provider] }, last_session: 1.hour.ago..Time.zone.now, online: false)
+
+    # Fetch all rooms associated with the recent meetings in advance.
+    recent_rooms = Room.where(meeting_id: recent_meetings.pluck(:meeting_id))
+                       .index_by(&:meeting_id)
```

Comment:
I'm not sure I understand what the purpose of this is - you're already fetching the meetings above, why re-fetch them?

Reply:
It's to reset the room.processing_recording

Reply:
Why can't you just do `meeting.update(recordings_processing....`?

Reply:
Right

Reply:
Thanks I had tunnel vision lol